tools/sklearn/main_macros.xml

```diff
@@ -106,49 +106,51 @@ def get_search_params(params_builder, in_pipeline=False):
 
   search_params = {}
 
-  if not in_pipeline:
-    for ep in params_builder['estimator_param_set']:
-      if ep['estimator_param'].strip() == '':
-        continue
-      lst = ep['estimator_param'].split(":")
-      assert (len(lst) == 2), "Error, make sure there is one and only one colon in search parameter input."
-      literal = lst[1].strip().replace("__sq__","'")\
-          .replace('__dq__','"').replace("__oc__", "{").replace("__cc__", "}")\
-          .replace("__ob__", "[").replace("__cb__", "]")
-      try:
-        ev = ast.literal_eval(literal)
-      except ValueError:
-        valid = ['np.', 'numpy.', 'scipy.']
-        for v in valid:
-          if literal.startswith(v):
-            ev = eval(literal)
-            break
-        else:
-          sys.exit("invalid search parameter input for %s" %ep['estimator_param'])
-      search_params[lst[0].strip()] = ev
-    return search_params
+  def in_whitelist(literal):
+    patterns = ['^(np|numpy|scipy)\.(random|stats)\.\w+\(.*\)$', '^(np|numpy)\.arange\([\s\d.,]*\)$' ]
```

Comment:
I guess concerns would come from `\(.*\)$`. The inside of the parenthesis is the parameters of the numpy/scipy method. I could explicitly list the valid characters of parameters, but I didn't since I realize galaxy has already done the job.

Please refer to:
https://github.com/galaxyproject/galaxy/blob/988cab4db7ef83714cf6f0c57f73f4255e3f075d/lib/galaxy/tools/parameters/sanitize.py#L47